src/index.ts

```diff
@@ -541,19 +542,24 @@ export class Pool extends (EventEmitter as new () => PoolEmitter) {
       if (id) {
         this.connectionQueueEventEmitter.emit(`connection_${id}`, client);
       } else if (this.options.idleTimeoutMillis > 0) {
-        client.idleTimeoutTimer = setTimeout(() => {
-          this._removeConnection(client);
+        client.idleTimeoutTimer = setTimeout((): void => {
+          // eslint-disable-next-line no-void
+          void this._removeConnection(client);
         }, this.options.idleTimeoutMillis);
 
         this.idleConnections.push(client);
         this.emit('connectionIdle');
       } else {
-        this._removeConnection(client);
+        await this._removeConnection(client);
       }
     };
 
-    client.errorHandler = (err: Error): void => {
-      this._removeConnection(client);
+    // we assign an async promise to a void
+    // because we don't want to enforce a promise signature
+    // on errorHandler
+    // also, EventEmitter.on("error") doesn't support () => Promise<void> (see the on("error"))
+    client.errorHandler = async (err: Error): Promise<void> => {
```

Comment:
Thoughts on leaving errorHandler return void?

```ts
client.errorHandler = (err: Error): void => {
  this._removeConnection(client).finally(() => this.emit('error', err, client));
}
```

Reply:
Makes sense if we want to keep `this._removeConnection`. Will do.